### ISSUE TITLE
14

### ISSUE THREAD

**Issue by mpektkd** [slp-ntua/slp-labs#17]
Για την εκπαιδευση του logistic regression νευρωνικου στο 14, μπορείτε να προτείνετε κάποια βιβλιοθήκη?

Επίσης, θα ήθελα να κάνω κάποιες διευκρινιστικές ερωτήσεις

1) 
![image](https://user-images.githubusercontent.com/62422421/98733029-99fe7480-23a8-11eb-87d3-b73a78febc72.png)
Εδώ που ζητάται να διαβάσουμε το corpus θα πρέπει να θεωρήσουμε corpus μόνο τις προτάσεις που υπάρχουν στα pos, neg για το train. Το λέω αυτό γιατί μετά πάμε πάλι κι χωρίζουμε το dataset με την συνάρτηση split?,,,Η πρέπει να κατεβάσουμε το Imdb corpus όπως το gutenberg?...ή και τα δυο το ενα για τα embeddings και το άλλο για τα NBOW?

2)Επίσης στο 3ο εργαστήριο ειπώθηκε ότι πρέπει να χρησιμοποιήσουμε τα word embeddings για να φτιαξουμε τα NBOW. Αυτό θα γίνει όπως στο 12?

**Comment by georgepar** [comment]
> Για την εκπαιδευση του logistic regression νευρωνικου στο 14, μπορείτε να προτείνετε κάποια βιβλιοθήκη?

https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.LogisticRegression.html

Τις άλλες 2 ερωτήσεις δεν καταλαβαίνω ακριβώς τι ρωτάς.

**Comment by mpektkd** [comment]
Απλα δεν καταλαβαινω τι πρεπει να κανω με τα δεδομενα που μας δινετε. Μας εχετε δωσει ενα ζιπ που μεσα εχει train test directories, και το καθενα pos neg. Οταν λετε στο todo να διαβασουμε το corpus τι εννοειται?

Για την  εκπαιδευση του logistic regression model ποια data θα χρησιμοποιησω. Και τα NBOW θα προκυψουν απο που.

Γενικα αν μπορειτε να πειτε λιγο τη διαδικασια θα ηταν πολυ χρησιμο.

**Comment by georgepar** [comment]
Μπορείς να δεις και εδώ την εισαγωγή που κάναμε στο ML / sklearn (https://github.com/slp-ntua/python-lab/blob/master/Lab%200.3%20Scikit-learn.ipynb)

Τα pos / neg είναι τα labels, οπότε με βάση αυτά φτιάχνεις το y.
Το Χ είναι η NBOW αναπαράσταση κάθε δείγματος.

Το NBOW το εξηγούμε στην εκφώνηση:

> Μια πρόταση μπορεί να αναπαρασταθεί ως ο μέσος όρος των w2v διανυσμάτων κάθε λέξης που περιέχει
(Neural Bag of Words). **Το NBOW είναι μια διανυσματική αναπαράσταση μιας πρότασης που προκύπτει
από το μέσο όρο των embeddings των λέξεων που την αποτελούν**. Σε αυτό το Βήμα θα κάνουμε χρήση
αυτών των αναπαραστάσεων για την ανάλυση συναισθήματος σε movie reviews

Τα train δεδομένα τα χρησιμοποιείς για την εκπαίδευση του μοντέλου και τα test για τον υπολογισμό του τελικού σκορ.

**Comment by mpektkd** [comment]
![image](https://user-images.githubusercontent.com/62422421/98866998-7f92cc80-2476-11eb-87d1-4466d371b41e.png)
εδω το corpus  που δινουμε για split προκυπτει απο τα pos neg του train μονο?

**Comment by georgepar** [comment]
Δεδομένου ότι το dataset έχει default train-test splits, δε θα χρειαζόταν αυτό.

Αλλά μπορεί να μη χωράει ολόκληρο το dataset στην μνήμη σας οπότε μπορεί να χρειαστεί να διαλέξετε ένα υποσύνολο του dataset να διαβάσετε και να το κάνετε train_test_split.

Οπότε το δοκιμάζεις και φορτώνεις όσα χωράνε.

**Comment by mpektkd** [comment]
Ωραια ευχαριστω

**Comment by mpektkd** [comment]
Εκπαιδευω το μοντελο, φορτωνοντας 10000(5000 pos, 5000 neg) κειμενα για training και 5000 για test(5000 pos, 5000 neg), αλλα δεν χρησιμοποιω την train_test_split, αφου ειναι χωρισμενα. Ειναι καλή αναλογία  η θα πρεπει να μειωσω το test?

**Comment by georgepar** [comment]
>  5000 για test(5000 pos, 5000 neg) 
κατι πήγε στραβά με το copy paste?

Νορμάλ ακούγεται.

**Comment by mpektkd** [comment]
ωραια...χρησιμοποιησα την sklearn για την εκπαιδευση και εκανα print με τη συναρτηση του εργαστηριου για να δω το οριο διαχωρισμου...οπου βλεπουμε οτι ειναι ολα μαζι τα NBOW στο επιπεδο(εχω χρησιμοποιησει TSNE για να μειωσω τη διασταση)...ειναι λογικο να γινει αυτο???(με βαση τα embeddings που εχουμε απο την 12) η πρεπε να ειναι διαφορετικα?


Η ιδια εικονα εξαγεται κι για τα goole news....
![image](https://user-images.githubusercontent.com/62422421/99147022-f7831180-2685-11eb-99ea-15e30de9d3a3.png)

**Comment by georgepar** [comment]
> Η ιδια εικονα εξαγεται κι για τα goole news....

Γενικά παίρνεις average από όλα τα embeddings των λέξεων για να κάνεις αναπαράσταση της πρότασης. Δεν νομίζω να δεις πολλά με ένα T-SNE.

Αν παίρνεις γύρω στο 70-80% accuracy θα έλεγα ότι είναι μια χαρά.

**Comment by LoniasGR** [comment]
> Τα pos / neg είναι τα labels, οπότε με βάση αυτά φτιάχνεις το y.
> Το Χ είναι η NBOW αναπαράσταση κάθε δείγματος.
> 
> Το NBOW το εξηγούμε στην εκφώνηση:
> 
> > Μια πρόταση μπορεί να αναπαρασταθεί ως ο μέσος όρος των w2v διανυσμάτων κάθε λέξης που περιέχει
> > (Neural Bag of Words). **Το NBOW είναι μια διανυσματική αναπαράσταση μιας πρότασης που προκύπτει
> > από το μέσο όρο των embeddings των λέξεων που την αποτελούν**. Σε αυτό το Βήμα θα κάνουμε χρήση
> > αυτών των αναπαραστάσεων για την ανάλυση συναισθήματος σε movie reviews
> 
> Τα train δεδομένα τα χρησιμοποιείς για την εκπαίδευση του μοντέλου και τα test για τον υπολογισμό του τελικού σκορ.

Άρα εμείς το κάθε review το θεωρούμε σαν μια πρόταση και φτιάχνουμε το NBOW της;

**Comment by georgepar** [comment]
Ναι ακριβώς

**Comment by LoniasGR** [comment]
Μια ακόμα ερώτηση από εμένα. 

Όταν τρέχω το `w2v_sentiment_analysis`, για τις default τιμές, το πρόγραμμα τερματίζεται από το λειτουργικό λόγω υπέρβασης των ορίων μνήμης, παρολο που έχω 16GB RAM στον υπολογιστή. (Φτάνει να καταναλώνει πάνω από 8 πριν τερματιστεί). Είναι αυτό πρόβλημα του κώδικα μου, ή υπάρχει κάποια τεχνική που θα έπρεπε να ακολουθήσω, ιδιαίτερα κατά την δημιουργία του corpus.

(Τρέχω σε WSL2, αν έχει σχέση).
